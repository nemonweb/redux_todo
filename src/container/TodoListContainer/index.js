import { connect } from 'react-redux'
import { toggleTodo } from '../../actions'
import TodoList from '../../components/TodoList/index'

const mapStateToProps = state => ({ todos: state.todos })

const mapDispatchToProps = dispatch => ({
  onTodoClick: id => {
    dispatch(toggleTodo(id))
  }
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoList)
