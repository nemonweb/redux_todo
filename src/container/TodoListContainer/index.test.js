import React from 'react'
import TodoListContainer from './index'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

describe('TodoListContainer', () => {
  const mockStore = configureStore([])
  const initialState = { todos: [{ completed: false, text: 'xxx' }] }
  const store = mockStore(initialState)

  it('should properly render self with children', () => {
    const wrapper = mount(
      <Provider store={store}>
        <TodoListContainer />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should properly work callbacks', () => {
    const wrapper = shallow(<TodoListContainer store={store} />)
    wrapper.props().onTodoClick()
    expect(store.getActions()).toEqual([{ type: 'TOGGLE_TODO' }])
  })
})
