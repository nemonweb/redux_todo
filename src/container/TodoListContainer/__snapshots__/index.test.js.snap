// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListContainer should properly render self with children 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TodoList)>
    <TodoList
      onTodoClick={[Function]}
      todos={
        Array [
          Object {
            "completed": false,
            "text": "xxx",
          },
        ]
      }
    >
      <ul>
        <TodoItem
          completed={false}
          index={0}
          key="0"
          onClick={[Function]}
          text="xxx"
        >
          <li
            onClick={[Function]}
          >
            xxx
             
            -
          </li>
        </TodoItem>
      </ul>
    </TodoList>
  </Connect(TodoList)>
</Provider>
`;
